src/libp2p/Libp2p.Protocols.Tls/TlsProtocols.cs

```diff
@@ -0,0 +1,178 @@
+using System.Buffers;
+using System.Net;
+using System.Net.Security;
+using Nethermind.Libp2p.Protocols.Quic;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Extensions.Logging;
+using System.Security.Cryptography;
+using Nethermind.Libp2p.Core;
+using Multiformats.Address;
+using Multiformats.Address.Protocols;
+using System.Text;
+
+namespace Nethermind.Libp2p.Protocols;
+
+public class TlsProtocol(MultiplexerSettings? multiplexerSettings = null, ILoggerFactory? loggerFactory = null) : IProtocol
+{
+    private readonly ECDsa _sessionKey = ECDsa.Create();
+    private readonly ILogger<TlsProtocol>? _logger = loggerFactory?.CreateLogger<TlsProtocol>();
+    private readonly ILogger<ChannelStream>? _logger1 = loggerFactory?.CreateLogger<ChannelStream>();
+
+    public SslApplicationProtocol? LastNegotiatedApplicationProtocol { get; private set; }
+
+    private readonly List<SslApplicationProtocol> _protocols = multiplexerSettings is null ?
+        new List<SslApplicationProtocol> { } :
+        !multiplexerSettings.Multiplexers.Any() ?
+        new List<SslApplicationProtocol> { } :
+        multiplexerSettings.Multiplexers.Select(proto => new SslApplicationProtocol(proto.Id)).ToList();
```

Comment:
multiplexerSettings is used by muxer protocols to register themselves before communication starts. But it may happen that TlsProtocol will be created before them, so the settings will be empty. So instead of instantly creating list of `SslApplicationProtocol`'s, could you create it in listenasync/dialasync methods when the needed instead. Will look not so efficient, but it will work at least